src/components/ElementBar/ColorPickerButton/extractFirstNonTransparentOrFirstColor.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2024 Nordeck IT + Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Elements } from '../../../state/types';
+
+/**
+ * Extracts the first non-transparent colour of the elements, if available.
+ * Otherwise returns "transparent" if elements are not empty.
+ * Otherwise returns "undefined".
+ *
+ * @param elements - Elements to search for a colour
+ * @returns first non-transparent colour or "transparent" or "undefined"
+ */
+export function extractFirstNonTransparentOrFirstColor(
+  elements: Elements,
```

Comment:
elements type should be `Element[]` because we don't need `ids` from `Elements` type for this function.
```suggestion
  elements: Element[],
```

Reply:
Updated